### DISCUSSION TITLE
Can we get il back from ilast after runninf transforms over it?

### DISCUSSION THREAD

**Discussion by 0x410c** [icsharpcode/ILSpy#3174] [General] [answered]
hey,

i was looking for an option to get il code back from ilast? is it possible?

**Comment by dgrunwald** [comment, marked as answer]
No; there's currently no ILAst-to-IL compiler.

You would need additional transforms in the opposite direction to lower high-level constructs (e.g. async/await) back to IL code; and then somehow connect the ILAst to an assembly-writing library.
